flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/table/util/KinesisStreamsConnectorOptionsUtils.java

```diff
@@ -148,13 +149,13 @@ public static class KinesisProducerOptionsMapper {
         public KinesisProducerOptionsMapper(
                 ReadableConfig tableOptions, Map<String, String> resolvedOptions) {
             this.tableOptions = tableOptions;
-            this.resolvedOptions = resolvedOptions;
+            this.resolvedOptions = new HashMap<>(resolvedOptions);
         }
 
         @VisibleForTesting
         public KinesisProducerOptionsMapper(Map<String, String> allOptions) {
```

Comment:
Change parameter `allOptions` to `resolvedOptions` to make it consistent?

Reply:
The constructor is used differently, the one with `allOPtions` should contain both `tableOptions` and `resolvedOptions`. you can see it is parsed into both in this constructor.